compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -4691,7 +4691,7 @@ private BType getIntersection(IntersectionContext intersectionContext, BType lhs
                     return symTable.semanticError;
                 }
 
-                return ImmutableTypeCloner.getEffectiveImmutableType(null, this, bType,
+                return ImmutableTypeCloner.getEffectiveImmutableType(bType.tsymbol.pos, this, bType,
```

Comment:
IMO, this is still not correct.

`bType` here could come from a different module.

Reply:
I changed it pass form calle methods, please look